authors/bruno-skvorc/polkadot/en_US/final.md

```diff
@@ -0,0 +1,186 @@
+
+The first and default execution environment will be eWASM, a neutered version of WASM flavored for Ethereum. The second one may very well be an EVM EE which would allow Ethereum 1 to continue living and breathing inside of Ethereum 2, perhaps on a dedicated shard. Developing custom EEs will be expensive and hard, so we don't expect to see many of them.
+
+eWASM will support smart contracts through languages that can compile to WASM like [Nimplay](https://github.com/status-im/nimplay/). EVM EE will support smart contracts just like Ethereum does today.
+
+In Polkadot, the relay chain does not have smart contract support. Given that there are no parachains yet, there is effectively nowhere to go if you want to develop a smart contract of any kind on Polkadot. However, Substrate - the framework on which the relay chain is built, and the framework which others will likely use to deploy parachains, has support for modules - blobs of plug-and-play code used to enhance your chain's runtime with new functionality. One such module is the Smart Contracts module which, when plugged into a parachain, can support smart contract execution. There are two caveats to such a model:
+
+1. When you want to just fire up a smart contract, you have to do it on a specific smart-contract parachain (though nothing prevents you from deploying the same thing on all smart contract parachains). Picking between them might be difficult - they can be custom-tweaked for specific use cases (privacy vs speed, for example).
+2. That smart contract chain has to communicate with other chains to be of any use, otherwise you're working in a silo.
+
+Given that Polkadot's entire story arc is the interconnectivity of these parachains, the communciation aspect is rather well developed. We mentioned this briefly in the ICMP section above, but here's a recap: the parachain has an egress (outgoing) and ingress (incoming) queue of messages. These messages are processed in order and happen on chain. But how can you be sure your destination chain executed the code you wanted to trigger by sending it a message? This is where SPREEs come in.
+
+Think of SPREEs - Shared Protected Runtime Execution Enclaves - as part of a "standard library" for parachains. The standard library will contain many different SPREEs, and they all live on the relay chain as WASM blobs of code that parachains can "opt in to". For example, say there is a SPREE which handles token transfers. If two parachains opt into this SPREE, they both get the built-in ability to transfer tokens. Thus, a message that says "transfer tokens" means the same thing to both chains, and they can reliably communicate. SPREEs have their own ICMP endpoints so they can receive messages and their own isolated storage.
```

Comment:
```suggestion
Think of SPREEs - Shared Protected Runtime Execution Enclaves - as part of a "standard library" for parachains. The standard library will contain many different SPREEs, and they all live on the relay chain as Wasm blobs of code that parachains can "opt in to". For example, say there is a SPREE which handles token transfers. If two parachains opt into this SPREE, they both get the built-in ability to transfer tokens. Thus, a message that says "transfer tokens" means the same thing to both chains, and they can reliably communicate. SPREEs have their own ICMP endpoints so they can receive messages and their own isolated storage.
```

Reply:
I think that you could go into more detail on what SPREEs are, but this is already pretty long so perhaps it could be done in a separate post.